src/com/goide/inspections/GoCommentStartInspection.java

```diff
@@ -0,0 +1,109 @@
+            (!(commentParent instanceof GoTopLevelDeclaration))) {
+          return;
+        }
+
+        String elementName = null;
+        PsiElement nextElement = PsiTreeUtil.skipSiblingsForward(comment, PsiWhiteSpace.class);
+        if ((nextElement instanceof GoNamedElement)) {
+          elementName = ((GoNamedElement)nextElement).getName();
+        }
+        else if (nextElement instanceof GoPackageClause) {
+          PsiElement identifier = ((GoPackageClause)nextElement).getIdentifier();
+          if (identifier == null) {
+            return;
+          }
+          elementName = identifier.getText();
+        }
+        else if (nextElement instanceof GoTypeDeclaration) {
+          if (((GoTypeDeclaration)nextElement).getTypeSpecList().size() == 0) {
+            return;
+          }
+          elementName = ((GoTypeDeclaration)nextElement).getTypeSpecList().get(0).getName();
+        }
+        else if (nextElement instanceof GoConstSpec) {
+          if (((GoConstSpec)nextElement).getConstDefinitionList().size() == 0) {
+            return;
+          }
+          elementName = ((GoConstSpec)nextElement).getConstDefinitionList().get(0).getName();
+        }
+        else if (nextElement instanceof GoConstDeclaration) {
+          if (((GoConstDeclaration)nextElement).getConstSpecList().size() == 0) {
+            return;
+          }
+          elementName = ((GoConstDeclaration)nextElement).getConstSpecList().get(0).getConstDefinitionList().get(0).getName();
+        }
+        else if (nextElement instanceof GoVarSpec) {
+          if (((GoVarSpec)nextElement).getVarDefinitionList().size() == 0) {
+            return;
+          }
+          elementName = ((GoVarSpec)nextElement).getVarDefinitionList().get(0).getName();
+        }
+        else if (nextElement instanceof GoVarDeclaration) {
+          if (((GoVarDeclaration)nextElement).getVarSpecList().size() == 0) {
+            return;
+          }
+          elementName = ((GoVarDeclaration)nextElement).getVarSpecList().get(0).getVarDefinitionList().get(0).getName();
+        }
+
+        if (elementName == null ||
+            (!(nextElement instanceof GoPackageClause)) && !StringUtil.isCapitalized(elementName)) {
+          return;
+        }
+
+        String commentText = comment.getText();
+        if (!correctCommentStart(commentText, elementName)) {
+          holder.registerProblem(comment, "Comment should start with '" + elementName + "'", ProblemHighlightType.WEAK_WARNING);
+        }
+        // +4 stands for //<space>Element_Name<space>
+        else if (commentText.length() <= elementName.length() + 4) {
+          holder.registerProblem(comment, "Comment should be meaningful or it should be removed", ProblemHighlightType.WEAK_WARNING);
+        }
+      }
+    };
+  }
+
+  private static boolean correctCommentStart(String commentText, String elementName) {
+    return commentText.startsWith("// " + elementName) ||
```

Comment:
I've also linked it in in the class comment, forgot to do it: https://github.com/golang/lint/blob/32a87160691b3c96046c0c678fe57c5bef761456/lint.go#L744